src/BeadingStrategy/CenterDeviationBeadingStrategy.cpp

```diff
@@ -20,37 +20,32 @@ CenterDeviationBeadingStrategy::CenterDeviationBeadingStrategy
 
 CenterDeviationBeadingStrategy::Beading CenterDeviationBeadingStrategy::compute(coord_t thickness, coord_t bead_count) const
 {
-    const coord_t minimum_line_width_split = optimal_width * wall_split_middle_threshold;
-    const coord_t minimum_line_width_add = optimal_width * wall_add_middle_threshold;
-
     Beading ret;
 
     ret.total_thickness = thickness;
     if (bead_count > 0)
     {
         // Set the bead widths
         ret.bead_widths = std::vector<coord_t>(static_cast<size_t>(bead_count), optimal_width);
-        const coord_t optimal_thickness = getOptimalThickness(bead_count);
-        const coord_t diff_thickness = thickness - optimal_thickness; //Amount of deviation. Either spread out over the middle 2 lines, or concentrated in the center line.
-        const size_t center_bead_idx = ret.bead_widths.size() / 2;
-        if (bead_count % 2 == 0) // Even lines
-        {
-            const coord_t inner_bead_widths = optimal_width + diff_thickness / 2;
-            if (inner_bead_widths < minimum_line_width_add)
-            {
-                return compute(thickness, bead_count - 1);
-            }
-            ret.bead_widths[center_bead_idx - 1] = inner_bead_widths;
-            ret.bead_widths[center_bead_idx] = inner_bead_widths;
-        }
-        else // Uneven lines
+        coord_t leftover_thickness = ret.total_thickness;
+        for (size_t bead_index = 0; bead_index <= bead_count / 2; ++bead_index)
         {
-            const coord_t inner_bead_widths = optimal_width + diff_thickness;
-            if (inner_bead_widths < minimum_line_width_split)
+            const size_t opposite_bead_index = bead_count - (1 + bead_index);
+            switch (opposite_bead_index - bead_index)
             {
-                return compute(thickness, bead_count - 1);
+            case 0: // Single bead in the middle:
+                ret.bead_widths[bead_index] = leftover_thickness;
+                break;
+            case 1: // Two beads in the middle:
+                ret.bead_widths[bead_index] = leftover_thickness / 2;
+                ret.bead_widths[opposite_bead_index] = leftover_thickness / 2;
+                break;
+            default: // Beads on the outside.
+                ret.bead_widths[bead_index] = optimal_width;
+                ret.bead_widths[opposite_bead_index] = optimal_width;
+                break;
```

Comment:
This is applied inconsistently throughout the engine. I personally prefer it unindented, since that puts it on the same level as an `if-else` would have put it.

(Update: We talked it out in slack and decided not to.)